src/tree-sitter-javascript/scanner.h

```diff
@@ -0,0 +1,333 @@
+// ----------------------------------------------------------
+//
+//
+//
+// Scanners are not extensible, so we have to copy them,
+//
+// To Update:
+//  - delete everything under the comment block
+//  - find and replace "tree_sitter_javascript" with "tree_sitter_javascript"
+//    - there should be 5 methods updated
+//    - %s/tree_sitter_javascript/tree_sitter_javascript/g
+//
+//
+//
+// ----------------------------------------------------------
+
+#include "tree_sitter/parser.h"
+
+#include <wctype.h>
+
+enum TokenType {
+    AUTOMATIC_SEMICOLON,
+    TEMPLATE_CHARS,
+    TERNARY_QMARK,
+    HTML_COMMENT,
+    LOGICAL_OR,
+    ESCAPE_SEQUENCE,
+    REGEX_PATTERN,
+    JSX_TEXT,
+};
+
+void *tree_sitter_javascript_external_scanner_create() { return NULL; }
```

Comment:
shouldn't these be tree_sitter_glimmer_javascript?

Reply:
maybe it doesn't matter?

Reply:
At the moment these aren’t in use, and duplicated version with `glimmer_javascript` are in `scanner.c`.

I see three options; leave them, call them, or remove them.

 I thought it would be nice to have the ability to just copy paste `scanner.c` from GitHub into `scanner.h`.  So maybe best to just call them from `scanner.c`?

Reply:
They are now called from scanner.c